src/sql/workbench/api/common/extHostDataProtocol.ts

```diff
@@ -324,6 +324,8 @@ export class ExtHostDataProtocol extends ExtHostDataProtocolShape {
 		if (this.uriTransformer) {
 			batchInfo.ownerUri = URI.from(this.uriTransformer.transformOutgoing(URI.parse(batchInfo.ownerUri))).toString(true);
 		}
+		this.messageRunner.cancel(); // clear batch messages before saying we completed the batch
```

Comment:
Maybe add a bit more to indicate that this is done to ensure the messages are printed in the correct order